.github/workflows/code-quality.yaml

```diff
@@ -0,0 +1,46 @@
+name: Code Quality checks for PRs
```

Comment:
looking at the deploy-functions-oidc repo, it looks like this can be done a lot easier:
https://github.com/cognitedata/deploy-functions-oidc/blob/a9ed5f7bec35c438e6790a94a8501c3ce76df004/.github/workflows/code-quality.yaml
`name: Code Quality checks for PRs

on:
  pull_request:

jobs:
  pre-commit:
    name: Pre-Commit checks
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-python@v2
      with:
        python-version: '3.8'
    - uses: pre-commit/action@v2.0.0`

Reply:
looks better :)
let's test

Reply:
Agree that the above is easier if you are only running pre-commit, as then, there is no need to setup poetry + dependencies. The job this was copied also runs pytest, bandit and safety, which requires the poetry+dependencies installed. 

**Note** the action pre-commit/action@v2.0.0 is deprecated.